src/middleware.ts

```diff
@@ -281,6 +281,8 @@ export const persist = <S extends State>(
         } else {
           set(deserializedStorageValue.state)
         }
+      } else {
+        await storage.setItem(name, await serialize({...get() || {}})
```

Comment:
This seems another issue from what I think we should fix firstly, but I am not sure if this is desired. At least I don't want to write to localStorage without change. I think this should be optional, and if async issues are solved, there's a workaround, and we will see what we want.

Reply:
the initial value of the store shouldn't be written in localStorage? Do you mean this?

for example 
```Javascript
const useStore = create((set) => ({
     friends: 10,
     setFriends: () => set({friends: 20})
}, {name: "my-storage"}))
```

so the localStorage will not store the value `friends: 10`, until you update the value by `setFriends`

Reply:
Yes, in my use case in mind, I wouldn't store the default value.

```js
const useStore = create(persist((set) => ({
     userName: "No Name",
     setUserName: (userName) => set({ userName })
}), { name: "my-user-setting" }))
```

In this case, I don't want to store "No Name" in localStorage.

Reply:
Hm, and your use case of storing default value seems already covered by `postRehydrationCallback`. So, it's already supported, I'd say.

Reply:
storing is optional right? 
and if someone wants to store, it should use the `postRehydrationCallback` callback right ?